plugins/chat/assets/javascripts/discourse/models/chat-message.js

```diff
@@ -83,6 +93,36 @@ export default class ChatMessage {
     this.bookmark = args.bookmark ? Bookmark.create(args.bookmark) : null;
   }
 
+  cook() {
+    const site = getOwner(this).lookup("service:site");
+
+    const markdownOptions = {
+      featuresOverride:
+        site.markdown_additional_options?.chat?.limited_pretty_text_features,
+      markdownItRules:
+        site.markdown_additional_options?.chat
+          ?.limited_pretty_text_markdown_rules,
+      hashtagTypesInPriorityOrder:
+        site.hashtag_configurations?.["chat-composer"],
+      hashtagIcons: site.hashtag_icons,
+    };
+
+    if (ChatMessage.cookFunction) {
+      this.cooked = ChatMessage.cookFunction(this.message);
+    } else {
+      generateCookFunction(markdownOptions).then((cookFunction) => {
```

Comment:
Just curious, why move this out of the initializer? If we have it here shouldn't we return the promise, which also makes `message.cook()` calls more complex since you always have to do `.then()`?

Reply:
I was having cases where actual code was run before the `generateCookFunction` was done and stopping the program at this point is not a good idea it would make the whole app slower. SO this design should be much more resilient and shouldn't need the` .then()`

Reply:
No problem, just thought I would check :+1: